### PR TITLE
e2e Record Splitter test scenarios

### PR DESCRIPTION
**Pull request by itsmekumari**
@sawantpritam Please review

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,24 +50,52 @@
       <artifactId>cdap-api</artifactId>
       <version>${cdap.version}</version>
       <scope>provided</scope>
+      <!--<exclusions>
```
Comment (sawantpritam) [inline pom.xml]: Remove commented exclusions

File: src/e2e-test/resources/pluginParameters.properties [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+dataset=test_automation
+serviceAccountAutoDetect=auto-detect
+csvFormat=csv
+
+serviceAccountType=filePath
+serviceAccount=auto-detect
+bqTruncateTable=true
+bqUpdateTableSchema=true
+
+## FILE-PLUGIN-PROPERTIES-START
```
Comment (sawantpritam) [inline src/e2e-test/resources/pluginParameters.properties]: Remove unused file properties

File: src/e2e-test/resources/testdata/file/CSV_DATATYPE_TEST_1.csv [hunk 3]
```diff
@@ -0,0 +1,5 @@
+id,name,yearofbirth,isdeleted,email,createddate,revenue,points,longdatatype,doubledatatype,date,null,BytesData
```
Comment (sawantpritam) [inline src/e2e-test/resources/testdata/file/CSV_DATATYPE_TEST_1.csv]: Remove unused testdata files

File: src/e2e-test/resources/cmek-config.properties [hunk 4]
```diff
@@ -0,0 +1,4 @@
+cmekGCS=projects/cdf-athena/locations/us/keyRings/gcs-cmek-test/cryptoKeys/test_key
```
Comment (sawantpritam) [inline src/e2e-test/resources/cmek-config.properties]: Remove this file as we dont have cmek test scenarios

File: src/e2e-test/java/io/cdap/plugin/utils/E2ETestConstants.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.utils;
+
+/**
+ * E2E test constants.
+ */
+public class E2ETestConstants {
+}
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/utils/E2ETestConstants.java]: Empty constants file can be removed.

File: src/e2e-test/java/io/cdap/plugin/recordsplitter/actions/RecordSplitterActions.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * RecordSplitter plugin step actions.
+ */
+public class RecordSplitterActions {
+  private static final Logger logger = LoggerFactory.getLogger(RecordSplitterActions.class);
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/recordsplitter/actions/RecordSplitterActions.java]: Remove unused logger

File: src/e2e-test/features/recordsplitter/BigQueryToRecordSplitter.feature [hunk 7]
```diff
@@ -0,0 +1,105 @@
+@RecordSplitter
+Feature: RecordSplitter transform - Verify BigQuery source data transfer using RecordSplitter transformation
+
+  @BQ_SOURCE_DELIMITED_TEST @BQ_SINK_TEST
+  Scenario: To verify data is getting transferred from BigQuery to BigQuery sink plugin successfully with Delimiter
```
Comment (sawantpritam) [inline src/e2e-test/features/recordsplitter/BigQueryToRecordSplitter.feature]: with Delimiter -> with record splitter

check all descriptions

**Review by sawantpritam — CHANGES_REQUESTED** [review]

**Comment by sawantpritam** [comment]
Add Override service account details steps in all scenarios

File: pom.xml (hunk 1 above)
Comment (itsmekumari) [reply pom.xml]: Removed.

File: src/e2e-test/resources/testdata/file/CSV_DATATYPE_TEST_1.csv (hunk 3 above)
Comment (itsmekumari) [reply src/e2e-test/resources/testdata/file/CSV_DATATYPE_TEST_1.csv]: Removed.

File: src/e2e-test/resources/pluginParameters.properties (hunk 2 above)
Comment (itsmekumari) [reply src/e2e-test/resources/pluginParameters.properties]: Removed.

File: src/e2e-test/resources/cmek-config.properties (hunk 4 above)
Comment (itsmekumari) [reply src/e2e-test/resources/cmek-config.properties]: Removed.

File: src/e2e-test/java/io/cdap/plugin/utils/E2ETestConstants.java (hunk 5 above)
Comment (itsmekumari) [reply src/e2e-test/java/io/cdap/plugin/utils/E2ETestConstants.java]: Removed.

File: src/e2e-test/java/io/cdap/plugin/recordsplitter/actions/RecordSplitterActions.java (hunk 6 above)
Comment (itsmekumari) [reply src/e2e-test/java/io/cdap/plugin/recordsplitter/actions/RecordSplitterActions.java]: Removed.

File: src/e2e-test/features/recordsplitter/BigQueryToRecordSplitter.feature (hunk 7 above)
Comment (itsmekumari) [reply src/e2e-test/features/recordsplitter/BigQueryToRecordSplitter.feature]: Changed.

**Comment by itsmekumari** [comment]
> Add Override service account details steps in all scenarios

Added the Override service account details steps in all scenarios where required.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+  @Before(order = 1, value = "@FILE_SOURCE_TEST")
+  public static void setFileSourceAbsolutePath() {
+    if (firstFileSourceTestFlag) {
+      PluginPropertyUtils.addPluginProp("recordSplitterGCSCsvFile", Paths.get(TestSetupHooks.class.getResource
+        ("/" + PluginPropertyUtils.pluginProp("recordSplitterGCSCsvFile")).getPath()).toString());
+      firstFileSourceTestFlag = false;
+    }
+  }
+
+  @Before(order = 1, value = "@FILE_SINK_TEST")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:55]: Unused

**Review by sawantpritam — CHANGES_REQUESTED** [review]

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 8 above)
Comment (itsmekumari) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:55]: Added a test case with File sink.

**Comment by sawantpritam** [comment]
> > Add Override service account details steps in all scenarios
> 
> Added the Override service account details steps in all scenarios where required.

Add  below method in TestSetupHook for override to work in macro scenario. 
```
@Before(order = 1)
  public static void overrideServiceAccountFilePathIfProvided()
```
 https://github.com/data-integrations/google-cloud/blob/c3d0fc99448f56a2761d40dee8a2efe84b9ae24a/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java#L70

**Comment by itsmekumari** [comment]
> Added the method in TestSetupHook

**Pull request closed without merging**